File: 83.remove-duplicates-from-sorted-list.js
/*
 * @lc app=leetcode id=83 lang=javascript
 *
 * [83] Remove Duplicates from Sorted List
 */

// 配列をLoopで処理して、次の配列に格納をしていくイメージ
// 次の配列とvalueが同じなら、スキップ
// と思ったけれど、headが配列ではないらしい。list nodeってなんだよ

// @lc code=start
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */

/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    let current = head;
    while (current !== null && current.next !== null) {
      if (current.val === current.next.val) {
        // 次のノードを、さらにその次のノードで置き換える
        current.next = current.next.next;
      } else {
        // 次のノードへ
        current = current.next
      }
    }
    return head;
};
// @lc code=end

File: 83.remove-duplicates-from-sorted-list.go
/*
 * @lc app=leetcode id=83 lang=golang
 *
 * [83] Remove Duplicates from Sorted List
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    current := head
		for current != nil && current.Next != nil {
			if current.Val == current.Next.Val {
				current.Next = current.Next.Next
			} else {
				current = current.Next
			}
		}
		return head
}
// @lc code=end

